### PR TITLE
Prepare documentation for v.0.11 of MeiliSearch

### PR DESCRIPTION
**Pull request by bidoubiwa**
## Prepare version v0.11 of MeiliSearch

Displayed below are the changelog's for the new version of MeiliSearch (the list can still be expanded). All checkboxes should be checked once the appropriate documentation has been merged into this branch. 
Once all the boxes are checked, the branch is eligible to be merged on master.

## Contribute

To add documentation to cover the changes in the new version, please make a pull request that merges on this branch. Once it has been merged on this branch the box will be checked.

If a change in the core engine **does not require documentation** it should be crossed.

## Changelog

- [x] Change the HTTP framework, moving from tide to actix-web meilisearch/meilisearch#601
    - Breaking on max payload size
- [x] ~Bump sentry version to 0.18.1~ (meilisearch/meilisearch#690)
- [x] Enable max payload size override (meilisearch/meilisearch#684)
    - Changes on usage
- [x] ~Disable sentry in debug (meilisearch/meilisearch#681)~
- [x] ~Better terminal greeting (meilisearch/meilisearch#680)~
- [x] ~Fix highlight misalignment (meilisearch/meilisearch#679)~
- [x] Add support for facet count (meilisearch/meilisearch#676)
- [x] Add support for faceted search (meilisearch/meilisearch#631)
- [x] Add support for configuring the lmdb map size (meilisearch/meilisearch#646, meilisearch/meilisearch#647)
- [x] ~Add exposed port for Dockerfile (meilisearch/meilisearch#654)~
- [x] ~Add sentry probe (meilisearch/meilisearch#664)~
- [x] ~Fix url trailing slash and double slash issues (meilisearch/meilisearch#659)~
- [x] Fix accept all Content-Type by default (meilisearch/meilisearch#653)
- [x] Return the error message from Serde when a deserialization error is encountered (meilisearch/meilisearch#661)
- [x] ~Fix NormalizePath middleware to make the dashboard accessible (meilisearch/meilisearch#695)~
- [x] ~Update sentry features to remove openssl (meilisearch/meilisearch#702)~
- [x] Add SSL support (meilisearch/meilisearch#669)
- [x] Rename fieldsFrequency into fieldsDistribution in stats (meilisearch/meilisearch#719)
- [x] more analytics (meilisearch/MeiliSearch/issues/720)

### CODE REVIEW

File: guides/advanced_guides/configuration.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+```bash
+$ ./meilisearch --db-path ./meilifiles --http-addr '127.0.0.1:7700'
+Server is listening on: http://127.0.0.1:7700
+```
+
+## Passing arguments via the environment variables
+
+The format of the environment variables is identical to the command line options with the exception that it is uppercased and `MEILI_` is added at the beginning.
+
+```
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:20]:
```suggestion
```bash
```

File: guides/advanced_guides/configuration.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Configuration
+
+Many options are available to configure a MeiliSearch Instance. Each of these options are added on MeiliSearch Instance launch.
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:3]:
```suggestion
Many options are available to configure a MeiliSearch instance. Each of these options is added on MeiliSearch instance launch.
```

File: guides/advanced_guides/configuration.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+Server is listening on: http://127.0.0.1:7700
+```
+
+## Options
+
+List of options:
+
+- [Path to database](/guides/advanced_guides/configuration.md#path-to-database)
+- [HTTP address](/guides/advanced_guides/configuration.md#http-address)
+- [Master Key](/guides/advanced_guides/configuration.md#master-key)
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:33]:
```suggestion
- [Master key](/guides/advanced_guides/configuration.md#master-key)
```

File: guides/advanced_guides/configuration.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+- [No Analytics](/guides/advanced_guides/configuration.md#no-analytics)
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:34]:
```suggestion
- [No analytics](/guides/advanced_guides/configuration.md#no-analytics)
```

File: guides/advanced_guides/configuration.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+- [Environment](/guides/advanced_guides/configuration.md#environment)
+- [Payload size limit](/guides/advanced_guides/configuration.md#payload-size-limit)
+
+### Path to database
+
+**Environment Variable**: `MEILI_DB_PATH`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:40]:
```suggestion
**Environment variable**: `MEILI_DB_PATH`
```

File: guides/advanced_guides/configuration.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+**CLI option**: `--db-path`
+
+Defines the location for the database files.
+
+**Default value**: `"./data.ms"`
+
+### HTTP Address
+
+**Environment Variable**: `MEILI_HTTP_ADDR`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:49]:
```suggestion
**Environment variable**: `MEILI_HTTP_ADDR`
```

File: guides/advanced_guides/configuration.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+**CLI option**: `--http-addr`
+
+The address the HTTP server will listen on.
+
+**Default value**: `"127.0.0.1:7700"`
+
+### Master Key
+
+**Environment Variable**: `MEILI_MASTER_KEY`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:58]:
```suggestion
**Environment variable**: `MEILI_MASTER_KEY`
```

File: guides/advanced_guides/configuration.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+**CLI option**: `--master-key`
+
+The master key allowing you to do everything on the server. If no master key is provided all routes will be accessible without keys. This is only possible if your are in `developement` environment. An error is thrown if you try to start MeiliSearch without master key when the environment is set to `production`.
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:61]:
```suggestion
The master key allowing you to do everything on the server. If no master key is provided all routes will be accessible without keys. This is only possible if you are in `development` environment. An error is thrown if you try to start MeiliSearch without any master key when the environment is set to `production`.
```

File: guides/advanced_guides/configuration.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+[Learn more about the permission and authentication in this guide.](/guides/advanced_guides/authentication.md)
+
+**Default Value**: `None`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:65]:
```suggestion
**Default value**: `None`
```

File: guides/advanced_guides/configuration.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+### No Analytics
+
+**Environment Variable**: `MEILI_NO_ANALYTICS`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:69]:
```suggestion
**Environment variable**: `MEILI_NO_ANALYTICS`
```

File: guides/advanced_guides/configuration.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+**CLI option**: `--no-analytics`
+
+Deactivates analytics. Analytics allow us to know how many users are using MeiliSearch, which versions and which platforms are used. This process is entirely anonymous.
+
+**Default value**: `false`
+
+### Environment
+
+**Environment Variable**: `MEILI_ENV`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:78]:
```suggestion
**Environment variable**: `MEILI_ENV`
```

File: guides/advanced_guides/configuration.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+- `Development`: the [master key](/guides/advanced_guides/authentication.md) is **optional**, and logs are output in "info" mode (_console output_).
+
+If the server is running in development mode more logs will be displayed, and the master key can be avoided which implies that there is no security on the updates routes.
+This is useful to debug when integrating the engine with another service.
+
+**Default value**: `development`
+
+### Payload Size Limit
+
+**Environment Variable**: `MEILI_HTTP_PAYLOAD_SIZE_LIMIT`
```
Comment (curquiza) [inline guides/advanced_guides/configuration.md:93]:
```suggestion
**Environment variable**: `MEILI_HTTP_PAYLOAD_SIZE_LIMIT`
```

File: guides/advanced_guides/installation.md [hunk 13]
```diff
@@ -90,56 +90,13 @@ $ ./target/release/meilisearch
 
 ::::
 
-## Usage
+## Environment variables and Options
```
Comment (curquiza) [inline guides/advanced_guides/installation.md]:
```suggestion
## Environment Variables and Options
```

File: guides/advanced_guides/installation.md [hunk 14]
```diff
@@ -90,56 +90,13 @@ $ ./target/release/meilisearch
...
-| `MEILI_NO_ANALYTICS` | --no-analytics | Deactivates analytics. Analytics allow us to know how many users are using MeiliSearch, which versions and which platforms are used. This process is entirely anonymous. |                  |
-| `MEILI_ENV`          | --env          | Defines the running environment of MeiliSearch. Can be set to `production` or `development`.                                                                             | "development"    |
-
-### Environments
-
-By default, MeiliSearch runs in `development` mode.
-
-- `Production`: the [master key](/guides/advanced_guides/authentication.md) is **mandatory**.
-- `Development`: the [master key](/guides/advanced_guides/authentication.md) is **optional**, and logs are output in "info" mode (_console output_).
+Here is the list of [all Environment variables and options](/guides/advanced_guides/configuration.md).
```
Comment (curquiza) [inline guides/advanced_guides/installation.md]:
```suggestion
Here is the list of [all environment variables and options](/guides/advanced_guides/configuration.md).
```

File: guides/main_concepts/documents.md [hunk 15]
```diff
@@ -28,6 +28,19 @@ When using the [route to add new documents](/references/documents.md#add-or-upda
 
 In order to be indexed, each **document must contain** [the primary key field](/guides/main_concepts/documents.md#primary-key).
 
+## Upload
+
+By default, MeiliSearch limits the size of `JSON` payload to 10Mb. This affects the upload of documents.
+To upload more document in one go, it is possible to [change the payload size](/guides/advanced_guides/configuration.md#payload-size-limit) limit during the setup of the MeiliSearch instance using the `http-payload-size-limit` option.
+
+```
```
Comment (curquiza) [inline guides/main_concepts/documents.md:36]:
```suggestion
```bash
```

File: guides/main_concepts/documents.md [hunk 16]
```diff
@@ -28,6 +28,19 @@ When using the [route to add new documents](/references/documents.md#add-or-upda
...
+./meilisearch http-payload-size-limit=100000000
```
Comment (curquiza) [inline guides/main_concepts/documents.md:37]:
```suggestion
$ ./meilisearch http-payload-size-limit=100000000
```

File: guides/advanced_guides/installation.md (hunk 13 above)
Comment (bidoubiwa) [reply guides/advanced_guides/installation.md]: I changed the title after our call.

File: guides/main_concepts/documents.md [hunk 17]
```diff
@@ -28,6 +28,20 @@ When using the [route to add new documents](/references/documents.md#add-or-upda
...
 
+## Upload
+
+By default, MeiliSearch limits the size of `JSON` payload to 10Mb. This affects the upload of documents.
+
+To upload more document in one go, it is possible to [change the payload size](/guides/advanced_guides/configuration.md#payload-limit-size) limit during the setup of the MeiliSearch instance using the `http-payload-size-limit` option.
+
+```bash
+$ ./meilisearch http-payload-size-limit=100000000
+```
```
Comment (Kerollmops) [inline guides/main_concepts/documents.md:39]:
```suggestion
To upload more document in one go, it is possible to [change the payload size limit](/guides/advanced_guides/configuration.md#payload-limit-size) during the setup of the MeiliSearch instance using the `http-payload-size-limit` option.

```bash
$ ./meilisearch --http-payload-size-limit=100000000
```
```

File: guides/advanced_guides/installation.md [hunk 18]
```diff
@@ -90,56 +90,15 @@ $ ./target/release/meilisearch
...
-                                         keys - `development`: Show logs in "info" mode + not mendatory to specify the
-                                         api keys [env: MEILI_ENV=]  [default: development]  [possible values:
-                                         development, production]
-        --http-addr <http-addr>          The address on which the http server will listen. [env: MEILI_HTTP_ADDR=]
-                                         [default: 127.0.0.1:7700]
-        --master-key <master-key>        The master key allowing you to do everything on the server. [env:
-                                         MEILI_MASTER_KEY=]
-        --no-analytics <no-analytics>    Do not send analytics to Meili. [env: MEILI_NO_ANALYTICS=]
-```
+Options are added on launch. Either through command line options or through environment variables.
```
Comment (MarinPostma) [inline guides/advanced_guides/installation.md:95]:
```suggestion
Options are added at launch. Either through command line options or through environment variables.
```

File: guides/advanced_guides/search_parameters.md [hunk 19]
```diff
@@ -66,6 +68,192 @@ If you want to get only **two** documents, set `limit` to `2`.
...
+
+If you have [set up faceted attributes](/guides/advanced_guides/settings.md#attributes-for-faceting), you can filter on [facets](/guides/advanced_guides/faceted_search.md) to narrow down your results based on criteria.
+
+`facetFilters=["facetName:facetValue"]` or `facetFilters=[["facetName:facetValue"]]`
+or a mix of both `facetFilters=["facetName1:facetValue1", ["facetName2:facetValue2"]]`
+
+- `["facetName1:facetValue1", ["facetName2:facetValue2"]]` (Array of array of strings or single strings, defaults to `null`)
+
+  Both types of array contain the facet names and values to filter on.
+  A valid array must be an array which contains either a list of strings or arrays of strings and can mix both (e.g. `["director:Mati Diop", ["genre:Comedy", "genre:Romance"]]`).
```
Comment (MarinPostma) [inline guides/advanced_guides/search_parameters.md:103]: which  is for non-defining clause, when it is essential, use that (https://www.grammarly.com/blog/which-vs-that/)
```suggestion
  A valid array must be an array that contains either a list of strings or arrays of strings and can mix both (e.g. `["director:Mati Diop", ["genre:Comedy", "genre:Romance"]]`).
```

File: .code-samples.meilisearch.yaml [hunk 20]
```diff
@@ -173,6 +173,19 @@ update_distinct_attribute_1: |-
...
+  $ curl \
+    -X GET 'http://localhost:7700/indexes/movies/settings/attributes-for-faceting'
+update_attributes_for_faceting_1: |-
+  $ curl \
+    -X GET 'http://localhost:7700/indexes/movies/settings/attributes-for-faceting' \
+    --data '[
+        "genre",
+        "director"
+    ]'
+reset_attributes_for_faceting_1: |-
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply .code-samples.meilisearch.yaml:186]: You'r right!

```suggestion
    -X GET 'http://localhost:7700/indexes/movies/settings/attributes-for-faceting' \
    --get \
    --data '[
        "genre",
        "director"
    ]'
reset_attributes_for_faceting_1: |-
```

File: guides/advanced_guides/search_parameters.md (hunk 19 above)
Comment (bidoubiwa) [reply guides/advanced_guides/search_parameters.md:103]: great info thanks

File: .code-samples.meilisearch.yaml [hunk 21]
```diff
@@ -173,6 +173,20 @@ update_distinct_attribute_1: |-
 reset_distinct_attribute_1: |-
   $ curl \
     -X DELETE 'http://localhost:7700/indexes/movies/settings/distinct-attribute'
+get_attributes_for_faceting_1: |-
+  $ curl \
+    -X GET 'http://localhost:7700/indexes/movies/settings/attributes-for-faceting'
+update_attributes_for_faceting_1: |-
+  $ curl \
+    -X GET 'http://localhost:7700/indexes/movies/settings/attributes-for-faceting' \
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:181]:
```suggestion
    -X POST 'http://localhost:7700/indexes/movies/settings/attributes-for-faceting' \
```

File: .code-samples.meilisearch.yaml [hunk 22]
```diff
@@ -428,3 +442,39 @@ getting_started_search_md: |-
...
+        ]
+    }'
+faceted_search_walkthrough_facet_filters_1: |-
+  $ curl --get 'http://localhost:7700/indexes/movies/search' \
+      --data-urlencode 'q=thriller' \
+      --data-urlencode 'facetFilters=[["genres:Horror", "genres:Mystery"], "director:Jordan Peele"]'
+faceted_search_walkthrough_facets_distribution_1: |-
+  $ curl --get 'http://localhost:7700/indexes/movies/search' \
+      --data-urlencode 'q=Batman' \
+      --data-urlencode 'facetsDistribution=["genres"]'
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:480]: Missing newline

File: .vuepress/public/sample-template.yaml [hunk 23]
```diff
@@ -75,3 +75,9 @@ search_guide_2: |-
 getting_started_create_index_md: |-
 getting_started_add_documents_md: |-
 getting_started_search_md: |-
+faceted_search_update_settings_1: |-
+faceted_search_facet_filters_1: |-
+faceted_search_facets_distribution_1: |-
+faceted_search_walkthrough_attributes_for_faceting_1: |-
+faceted_search_walkthrough_facet_filters_1: |-
+faceted_search_walkthrough_facets_distribution_1: |-
```
Comment (Kerollmops) [inline .vuepress/public/sample-template.yaml:83]:
```suggestion
faceted_search_walkthrough_facets_distribution_1: |-

```

File: guides/advanced_guides/search_parameters.md [hunk 24]
```diff
@@ -66,6 +68,156 @@ If you want to get only **two** documents, set `limit` to `2`.
...
+- `[<facetName>, <facetName>, ...]` (Array of strings)
+
+  An array of strings that contains the facets for which to retrieve the matching count. The number of remaining candidates for each specified facet is returned.
+  If a facet name doesn't exist, it will be ignored.
+
+- `"*"`
+
+  The `*` character can also be used. In that case, a count for all facets is returned.
+
+[Learn more about `facetsDistribution` in the search parameters](/guides/advanced_guides/search_parameters.md#the-facets-distribution)
```
[the remark this answers is not in the document]
Comment (curquiza) [reply guides/advanced_guides/search_parameters.md:175]:
```suggestion
[Learn more about faceted search in the dedicated guide](/guides/advanced_guides/faceted_search.md)
```

**Comment by curquiza** [comment]
I was in API Reference, in the `Search` part. There is no example with `facetsDistribution`, so I clicked on `facetsDistribution` in the table.
I felt a little bit frustrated not to find any example in this redirection (advanced guides -> search parameters).
I had to click again on `Learn more about facetsDistribution in the search parameters` to finally have an example. (To be more accurate, the link does not work currenlty, but that's not the point.)


Could we put an example in advanced guides -> search parameters -> facetsDistribution?
I **really** like the Search parameters section in Advanced Guides. That's personally the part of the docs I use the most with the API References. And I don't really like being lost in other advanced guides when I only search for an example.

What do you think @bidoubiwa?

**Comment by bidoubiwa** [comment]
Examples are present but they do not appear as the code-samples on the master branch is not up to date! If you want to see the examples please look at the titles inside the code-samples.meilisearch.com file or by cloning and `yarn dev` as in development mode the samples are retrieved from the local directory

**Comment by curquiza** [comment]
Ok, thanks for the precision!! @bidoubiwa 

And what do you think about my second comment?

> I was in API Reference, in the Search part. There is no example with facetsDistribution, so I clicked on facetsDistribution in the table.
I felt a little bit frustrated not to find any example in this redirection (advanced guides -> search parameters).
I had to click again on Learn more about facetsDistribution in the search parameters to finally have an example. (To be more accurate, the link does not work currenlty, but that's not the point.)

> Could we put an example in advanced guides -> search parameters -> facetsDistribution?
I really like the Search parameters section in Advanced Guides. That's personally the part of the docs I use the most with the API References. And I don't really like being lost in other advanced guides when I only search for an example.

File: guides/advanced_guides/search_parameters.md [hunk 25]
```diff
@@ -66,6 +68,156 @@ If you want to get only **two** documents, set `limit` to `2`.
...
+If the `facetsDistribution` parameter has been set, the returned results will contain **two additional fields**:
+
+- `facetsDistribution`: The number of remaining candidates for each specified facet.
+
+- `exhaustiveFacetsCount`:
+  Returns `true` if the count in each facet value is **exhaustive** (exact count for each facet value).
+  Otherwise, returns `false` if this count is **approximative** (approximative count for each facet value).
+  The approximative facet count happens when there are too many documents in too many different facet values. In which case, MeiliSearch stops the distribution count to prevent considerably slowing down the request.
+
+[Learn more about facet distribution in the dedicated guide](/guides/advanced_guides/faceted_search.md)
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply guides/advanced_guides/search_parameters.md:188]: This is the facet distribution category but there was a duplicate in links. I removed one of them.

File: guides/advanced_guides/search_parameters.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply guides/advanced_guides/search_parameters.md:175]: I'm resolving this because there is a duplicate in the same category.

**Pull request merged**